src/garage/tf/models/categorical_lstm_model.py

```diff
@@ -0,0 +1,137 @@
+                 output_nonlinearity=None,
+                 output_w_init=tf.glorot_uniform_initializer(),
+                 output_b_init=tf.zeros_initializer(),
+                 hidden_state_init=tf.zeros_initializer(),
+                 hidden_state_init_trainable=False,
+                 cell_state_init=tf.zeros_initializer(),
+                 cell_state_init_trainable=False,
+                 forget_bias=True,
+                 layer_normalization=False):
+        super().__init__(
+            output_dim=output_dim,
+            hidden_dim=hidden_dim,
+            name=name,
+            hidden_nonlinearity=hidden_nonlinearity,
+            hidden_w_init=hidden_w_init,
+            hidden_b_init=hidden_b_init,
+            recurrent_nonlinearity=recurrent_nonlinearity,
+            recurrent_w_init=recurrent_w_init,
+            output_nonlinearity=output_nonlinearity,
+            output_w_init=output_w_init,
+            output_b_init=output_b_init,
+            hidden_state_init=hidden_state_init,
+            hidden_state_init_trainable=hidden_state_init_trainable,
+            cell_state_init=cell_state_init,
+            cell_state_init_trainable=cell_state_init_trainable,
+            forget_bias=forget_bias,
+            layer_normalization=layer_normalization)
+
+    def network_output_spec(self):
+        """Network output spec.
+
+        Returns:
+            list[str]: Name of the model outputs, in order.
+
+        """
+        return [
+            'all_output', 'step_output', 'step_hidden', 'step_cell',
+            'init_hidden', 'init_cell', 'dist'
+        ]
+
+    # pylint: disable=arguments-differ
+    def _build(self,
+               state_input,
+               step_input,
+               step_hidden,
+               step_cell,
+               name=None):
+        """Build model.
+
+        Args:
+            state_input (tf.Tensor): Entire time-series observation input.
+            step_input (tf.Tensor): Single timestep observation input.
+            step_hidden (tf.Tensor): Hidden state for step.
+            step_cell (tf.Tensor): Cell state for step.
+            name (str): Name of the model, also the name scope.
+
+        Returns:
+            tf.Tensor: Entire time-series outputs.
+            tf.Tensor: Step output.
+            tf.Tensor: Step hidden state.
+            tf.Tensor: Step cell state.
+            tf.Tensor: Initial hidden state.
+            tf.Tensor: Initial cell state.
+            tfp.distributions.OneHotCategorical: Distribution.
+
+        """
+        (prob, step_output, step_hidden, step_cell, init_hidden,
+         init_cell) = super()._build(state_input,
+                                     step_input,
+                                     step_hidden,
+                                     step_cell,
+                                     name=name)
+        dist = tfp.distributions.OneHotCategorical(prob)
+        return (prob, step_output, step_hidden, step_cell, init_hidden,
```

Comment:
why return `prob` when i can get it from `dist`?

Reply:
fixed, now it doesn't return prob